### PR TITLE
WT-8223 Enhance test/checkpoint to randomly operate with high eviction rate

### PR DESCRIPTION
**Pull request by clarissecheah**

### CODE REVIEW

File: test/checkpoint/test_checkpoint.c [hunk 1]
```diff
@@ -261,9 +261,12 @@ wt_connect(const char *config_open)
     static WT_EVENT_HANDLER event_handler = {
       handle_error, handle_message, NULL, NULL /* Close handler. */
     };
+    WT_RAND_STATE rnd;
     int ret;
     char config[512];
     char timing_stress_config[512];
+    bool fast_eviction_target = false;
```
Comment (jiechenbo) [inline test/checkpoint/test_checkpoint.c:268]: Place the initialisation below. WiredTiger likes to place the initialisation of variables after the declaration phase.

File: test/checkpoint/test_checkpoint.c [hunk 2]
```diff
@@ -261,9 +261,12 @@ wt_connect(const char *config_open)
...
+    bool fast_eviction_trigger = false;
```
Comment (jiechenbo) [inline test/checkpoint/test_checkpoint.c:269]: Place the initialisation below. WiredTiger likes to place the initialisation of variables after the declaration phase.

File: test/checkpoint/test_checkpoint.c [hunk 3]
```diff
@@ -278,21 +281,33 @@ wt_connect(const char *config_open)
           g.checkpoint_slow_timing_stress ? "checkpoint_slow" : ""));
     }
 
+    /*
+     * Randomly decide on the eviction rate (fast or default).
+     */
+
+    if ((__wt_random(&rnd) % 15) % 2 == 0) {
```
Comment (jiechenbo) [inline test/checkpoint/test_checkpoint.c:288]: Personally I would prefer the method of adding an extra configuration of `-e` to set the eviction fast or default. Let me know what you think around this. Also @kommiharibabu what are your opinions?

**Review by jiechenbo — COMMENTED** [review]
Just some changes!

File: test/checkpoint/test_checkpoint.c (hunk 2 above)
Comment (clarissecheah) [reply test/checkpoint/test_checkpoint.c]: Sure thing, just did that, thanks!

File: test/checkpoint/test_checkpoint.c [hunk 4]
```diff
@@ -278,21 +281,33 @@ wt_connect(const char *config_open)
...
+        fast_eviction_trigger = fast_eviction_target = true;
```
Comment (jiechenbo) [inline test/checkpoint/test_checkpoint.c:289]: If we are planning to go through having an option of doing `randomness`, it seems that having only one variable like `fast_eviction` is enough.

Also you can remove the braces for one line if statements

Comment (clarissecheah) [reply test/checkpoint/test_checkpoint.c]: Yep, thanks for pointing that out. I've changed it as you suggested.

File: test/checkpoint/test_checkpoint.c [hunk 5]
```diff
@@ -298,18 +296,18 @@ wt_connect(const char *config_open)
...
+          progname, fast_eviction ? 5 : 20, fast_eviction ? 1 : 5, timing_stress_config,
+          g.debug_mode ? DEBUG_MODE_CFG : "", config_open == NULL ? "" : ",",
           config_open == NULL ? "" : config_open));
-    else {
+    else
         testutil_check(__wt_snprintf(config, sizeof(config),
           "create,cache_cursors=false,statistics=(fast),statistics_log=(json,wait=1),log=(enabled),"
           "error_prefix=\"%s\",cache_size=1G, eviction_dirty_trigger=%i, "
           "eviction_dirty_target=%i,%s%s%s%s",
-          progname, fast_eviction_trigger ? 5 : 20, fast_eviction_target ? 1 : 5,
```
Comment (jiechenbo) [inline test/checkpoint/test_checkpoint.c]: Having eviction dirty target as 1 might be a bit scary. From my understanding the eviction update target is around 2 and is bigger. @kommiharibabu can you confirm?

**Review by jiechenbo — APPROVED** [review]
LGTM! Just have one question that needs to be addressed.

File: test/checkpoint/test_checkpoint.c (hunk 5 above)
Comment (kommiharibabu) [reply test/checkpoint/test_checkpoint.c]: Yes, by default `eviction_update_target` is half of `eviction_dirty_target`. The `eviction_update_target` also gets changed automatically with `eviction_dirty_target` unless you set both of them separately.

Comment (clarissecheah) [reply test/checkpoint/test_checkpoint.c]: Right, should I be changing the eviction_dirty_target to 2 then? I noticed that in` conn_api.c`, the `__conn_close` function changes the `eviction_dirty_trigger = 1.0 `and `eviction_dirty_target =0.1` to encourage eviction threads to clear the dirty content, so not as sure what a suitable value would be here.

File: test/checkpoint/test_checkpoint.c (hunk 3 above)
Comment (clarissecheah) [reply test/checkpoint/test_checkpoint.c]: Correct me if I'm wrong @kommiharibabu but I think the last time we had a conversation around this, we discussed that having it randomly was workable to achieve the desired result and takes less time to implement.

Comment (kommiharibabu) [reply test/checkpoint/test_checkpoint.c]: Yes, I prefer to go with the random approach as this way the high eviction rate is tested with all the existing options. That doesn't mean I don't like the external option approach, but I don't think it is required.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**